src/client/dfs/duns.c

```diff
@@ -0,0 +1,236 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * DAOS Unified namespace functionality.
+ */
+
+#define D_LOGFAC	DD_FAC(duns)
+
+#include <dirent.h>
+#include <sys/xattr.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <daos/common.h>
+#include "daos_types.h"
+#include "daos_api.h"
+#include "daos_fs.h"
+#include "daos_uns.h"
+
+#define DUNS_XATTR_NAME		"user.daos"
+#define DUNS_MAX_XATTR_LEN	150
+#define DUNS_XATTR_FMT		"DAOS.%s://%36s/%36s/%s/"
```

Comment:
Bruno will have to add code to this file to detect when the backend filesystem is lustre and support the uns so that the special LOV/LVM EA are used instead.

Reply:
Ok, understood. But starting to think of it, I wonder how do you both think it should work ? Should I use liblustreapi (and thus [statically ?] depend/link on/with it or dynamically/dlopen() load it ?) because it must be avail as a Lustre FS has been detected (using statfs() right?), or make it raw using ioctl()s ??

Reply:
I think we should use the liblustreapi, but i really don't want to add a dependency on Lustre when building DAOS, so we might have to use dlopen as you suggested.